src/backend/base/langflow/base/data/utils.py

```diff
@@ -120,79 +126,224 @@ def read_text_file(file_path: str) -> str:
+    except Exception as exc:
         if not silent_errors:
-            msg = f"Error loading file {file_path}: {e}"
-            raise ValueError(msg) from e
-        return None
+            msg = f"Could not parse xml file {file_path}: {exc}"
+            raise ValueError(msg) from exc
+
+    return None
+
+
+def convert_pdf_to_text(
+    file_path: str,
+    *,
+    silent_errors: bool = False,
+    do_ocr: bool = False,
+    ocr_lang: list[str] | None = None,
+    do_table_structure: bool = False,
+    do_picture_classification: bool = False,
+) -> Data | None:
+    try:
+        # fall back to english, pick your language here: https://www.jaided.ai/easyocr/
+        if ocr_lang is None:
+            ocr_lang = ["en"]
+
+        # OPTIMIZAION: could be better to "cache" the converters ...
+        pipeline_options = PdfPipelineOptions()
+        # pipeline_options.images_scale = 2
+        pipeline_options.generate_page_images = False
+        pipeline_options.do_picture_classification = do_picture_classification
+        pipeline_options.do_ocr = do_ocr
+        pipeline_options.do_table_structure = do_table_structure
+        pipeline_options.table_structure_options.do_cell_matching = False
+        pipeline_options.ocr_options.lang = ocr_lang
+        # pipeline_options.accelerator_options = AcceleratorOptions(num_threads=1, device=AcceleratorDevice.AUTO)
+
+        converter = DocumentConverter(
+            format_options={
+                InputFormat.PDF: PdfFormatOption(pipeline_options=pipeline_options),
+            }
+        )
+
+        result = converter.convert(source=Path(file_path))
+        markdown_text = result.document.export_to_markdown(
+            image_mode=ImageRefMode.PLACEHOLDER,
+            image_placeholder="<!-- image -->",
+        )
+
+        return Data(data={"file_path": file_path, "text": markdown_text})
+
+    except Exception as exc:
+        if not silent_errors:
+            msg = f"Could not parse pdf file {file_path}: {exc}"
+            raise ValueError(msg) from exc
+
+    return None
+
+
+def convert_img_to_text(
+    file_path: str,
+    *,
+    silent_errors: bool = False,
+    ocr_lang: list[str] | None = None,
+    do_table_structure: bool = False,
+    do_picture_classification: bool = False,
+) -> Data | None:
+    try:
+        # fall back to english, pick your language here: https://www.jaided.ai/easyocr/
+        if ocr_lang is None:
+            ocr_lang = ["en"]
+
+        # OPTIMIZAION: could be better to "cache" the converters ...
+        pipeline_options = PdfPipelineOptions()
+        # pipeline_options.images_scale = 2
+        pipeline_options.generate_page_images = False
+        pipeline_options.do_picture_classification = do_picture_classification
+        pipeline_options.do_ocr = True
+        pipeline_options.do_table_structure = do_table_structure
+        pipeline_options.table_structure_options.do_cell_matching = False
```

Comment:
I would comment this, we do want cell matching as default.

Reply:
done!